azureeyemodule/app/model/ssd.cpp

```diff
@@ -12,6 +12,7 @@
 #include <opencv2/gapi/infer.hpp>
 #include <opencv2/gapi/streaming/desync.hpp>
 #include <opencv2/highgui.hpp>
+#include <opencv2/gapi/render.hpp>
```

Comment:
Interesting. We have previously been rendering the inferences using traditional OpenCV library calls, rather than incorporating the rendering into the G-API.

I don't care which one we do, but we should do it the same way for all of the models. That being the case, it is way easier to just keep doing the way we've been doing it. Is there a reason to change to using G-API for rendering? I'm guessing it is not as easy or intuitive but may be more efficient.

Reply:
This is a long story. Let me try my best to make it clear here.

If you take a look at the custom BBoxes in ssd.cpp, namely, G_API_OP(BBoxes, <GPrims(GDetections)>, "sample.custom.b-boxes"), it's actually empty in the function body. So actually it's not rendering in G-API, but still with the traditional OpenCV library. Therefore, in the compile_cv_graph_uvc_video in objectdetection.cpp, I still call handle_bgr_output(). 

But why I used the rendering? It's because from the latest Intel drop 016 sample, they suggested to use such implementation to get the raw frame from a UVC camera or video file. On the other hand, in the original compile_cv_graph(), we desync'd the img_ts from img, which is
```
cv::GMat img = cv::gapi::copy(cv::gapi::streaming::desync(preproc));
cv::GOpaque<int64_t> img_ts = cv::gapi::streaming::timestamp(img);
```

but it's not needed when using uvc camera and video file.

Reply:
Hmm. Okay. Yes, let's make sure that we have a uniform way of doing it.

My preferences in order of what I would be most happy with, to what I would grumble about:

1. Have all object detectors use a common super-class's G-API graph and associated methods, regardless of input source. It sounds like this would require a few things: 1) dealing with the fact that H.264 is not output from anything but the MIPI camera (seems like we could get around that with an if statement) and 2) dealing with the timestamps; I think we will need the timestamps for the graphs when USB camera is the input, since we will still have the same problem with slow networks that we have with the MIPI - that the inferences get out of sync with the camera frames. We may also need to incorporate the inference thingy into the pipeline, but it is still not clear to me why.
2. Have all object detectors split their graphs into the MIPI one and the USB/file one, but they all use the same ones in the super class.
3. Have all models have a completely unique version of USB/file graph. Sad face.

I'd like to resolve this before merging in.

Reply:
After further discussions, we will keep the current PR as is